ui/apps/platform/cypress/integration/violations/violationsSearchFilter.test.ts

```diff
@@ -0,0 +1,116 @@
+import withAuth from '../../helpers/basicAuth';
+import { selectFilteredWorkflowView, visitViolations } from './Violations.helpers';
+
+function selectCompoundSearchFilterAttribute(attribute: string) {
+    cy.get('button[aria-label="compound search filter attribute selector toggle"]').click();
+    cy.get(
+        `ul[aria-label="compound search filter attribute selector menu"] button:contains("${attribute}")`
+    ).click();
+}
+
+function selectFirstAutocompleteSearchOption() {
+    cy.get(
+        'div[aria-labelledby="Filter results menu toggle"] button[aria-label="Menu toggle"]'
+    ).click();
+    cy.get(`ul[aria-label="Filter results select menu"] li:nth(0) button`).click();
+}
+
+function verifyTableCellsMatchFilterChip(cellSelector) {
+    cy.get('.search-filter-chips .pf-v5-c-chip .pf-v5-c-chip__text')
+        .invoke('text')
+        .then((textValue) => {
+            cy.get('table tbody tr').each(($row) => {
+                cy.wrap($row).find(cellSelector).invoke('text').should('equal', textValue);
+            });
+        });
+}
+
+describe('Violations - Search Filter', () => {
+    withAuth();
+
+    beforeEach(() => {
+        // clear the filters before every test
+        cy.document().then((doc) => {
+            const button = Cypress.$('button:contains("Clear filters")', doc);
+            if (button.length > 0) {
+                // if button exists, click it
+                cy.wrap(button).click();
+            }
+        });
+    });
+
+    it('should filter violations by policy name', () => {
+        visitViolations();
+
+        selectFilteredWorkflowView('Full view');
+
+        // filter by the first policy name autocomplete option
+        selectFirstAutocompleteSearchOption();
+
+        cy.intercept('GET', '/v1/alerts?query=*').as('getViolations');
+        cy.wait('@getViolations');
+
+        verifyTableCellsMatchFilterChip('td[data-label="Policy"] a');
+    });
+
+    it('should filter violations by policy category', () => {
```

Comment:
This one seems to fail still. There's a delay between the time an API response is received and when the table actually updates the rows. Since this table isn't using the new TBodyUnified component, I may need to do something else here. Maybe comparing the results text would help. I can keep track of what it was before the filter and once it changes, then we can verify that the table rows are filtered 🤔

Reply:
Since this doesn't have a good visual indicator of a loading state with `TBodyUnified`, it might be a good candidate for using either `interactAndWaitForResponses` to ensure the response is returned or `interceptAndWatchRequests` to do the same over multiple distinct events that each trigger a network call.